From 663d1a23a67c6d1af18de53c998cabc0c376be18 Mon Sep 17 00:00:00 2001
From: danielnachun <daniel.nachun@gmail.com>
Date: Mon, 26 Feb 2024 18:58:30 -0800
Subject: [PATCH] allow newer Guile

---
 agen5/guile-iface.h | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/agen5/guile-iface.h b/agen5/guile-iface.h
index af53f85..9fb137d 100644
--- a/agen5/guile-iface.h
+++ b/agen5/guile-iface.h
@@ -9,16 +9,13 @@
 # error AutoGen does not work with this version of Guile
   choke me.
 
-#elif GUILE_VERSION < 203000
+#else
 # define AG_SCM_IS_PROC(_p)           scm_is_true( scm_procedure_p(_p))
 # define AG_SCM_LIST_P(_l)            scm_is_true( scm_list_p(_l))
 # define AG_SCM_PAIR_P(_p)            scm_is_true( scm_pair_p(_p))
 # define AG_SCM_TO_LONG(_v)           scm_to_long(_v)
 # define AG_SCM_TO_ULONG(_v)          ((unsigned long)scm_to_ulong(_v))
 
-#else
-# error unknown GUILE_VERSION
-  choke me.
 #endif
 
 #endif /* MUTATING_GUILE_IFACE_H_GUARD */
-- 
2.43.0

