From ec5b0112dd451c66c43c744a35602829be1e72ea Mon Sep 17 00:00:00 2001
From: Daniel Nachun <daniel.nachun@gmail.com>
Date: Thu, 13 Mar 2025 14:21:57 -0700
Subject: [PATCH] make shared library

---
 Makefile | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 478aec8..1741e31 100644
--- a/Makefile
+++ b/Makefile
@@ -19,6 +19,11 @@ SUBDIRS=	.
 ifeq ($(shell uname -s),Linux)
 	LIBS += -lrt
 endif
+ifeq ($(shell uname -s),Linux)
+	LFLAGS = -shared
+else
+	LFLAGS = -dynamiclib -Wl,-install_name,@rpath/libbwa.dylib
+endif
 
 .SUFFIXES:.c .o .cc
 
@@ -27,14 +32,14 @@ endif
 
 all:$(PROG)
 
-bwa:libbwa.a $(AOBJS) main.o
+bwa:libbwa$(SHLIB_EXT) $(AOBJS) main.o
 		$(CC) $(CFLAGS) $(DFLAGS) $(AOBJS) main.o -o $@ -L. -lbwa $(LIBS)
 
-bwamem-lite:libbwa.a example.o
+bwamem-lite:libbwa$(SHLIB_EXT) example.o
 		$(CC) $(CFLAGS) $(DFLAGS) example.o -o $@ -L. -lbwa $(LIBS)
 
-libbwa.a:$(LOBJS)
-		$(AR) -csru $@ $(LOBJS)
+libbwa$(SHLIB_EXT):$(LOBJS)
+		$(CC) $(CFLAGS) $(DFLAGS) $(LFLAGS) -o $@ -lz $(LOBJS)
 
 clean:
 		rm -f gmon.out *.o a.out $(PROG) *~ *.a
-- 
2.47.1

