diff --git a/libStatGen/general/Makefile b/libStatGen/general/Makefile
index ed394426b7d29b9c02e5225728a55d06518a66d2..9e2b7096c50ae888353f1dbb3230d509a35a044b 100755
--- a/libStatGen/general/Makefile
+++ b/libStatGen/general/Makefile
@@ -8,10 +8,11 @@ ifeq ($(UNAME), Linux)
   # an error, but allow unused results and variables for the
   # time being.
   #
-  USER_WARNINGS ?= -Werror $(shell if [ X$(CCVERSION) \> X4.2.0 ] ; then echo " -Wno-strict-overflow" ; fi)
+#  USER_WARNINGS ?= -Werror $(shell if [ X$(CCVERSION) \> X4.2.0 ] ; then echo " -Wno-strict-overflow" ; fi)
 #-Wno-strict-overflow
 # -Wno-unused-variable $(shell if [ X$(CCVERSION) \> X4.2.0 ] ; then echo " -Wno-unused-result" ; fi)
 endif
+
 ifeq ($(UNAME), Darwin)
   # Mac OS, nothing to add.
   # -Wall contain -Wunused-variable and -Wunused-result from 'man gcc'
diff --git a/libStatGen/vcf/VcfRecordDiscardRules.h b/libStatGen/vcf/VcfRecordDiscardRules.h
index 1cc6fa7b18b7ded45bb078fded49befd3a34d5c2..2b532972580dc8314308fdc69dd65173f81020aa 100755
--- a/libStatGen/vcf/VcfRecordDiscardRules.h
+++ b/libStatGen/vcf/VcfRecordDiscardRules.h
@@ -19,7 +19,14 @@
 #define __VCF_RECORD_DISCARD_RULES_H__
 
 #include <vector>
+
+// added by jy for comilation bug with newer version gcc: ‘unordered_set’ in namespace ‘std’ does not name a template type
+#ifdef __GXX_EXPERIMENTAL_CXX0X__
+#include <unordered_set>
+#else
 #include <set>
+#endif
+
 #include <string>
 #include "VcfHeader.h"
 
@@ -51,14 +58,14 @@ class VcfRecordDiscardRules
     /// in the passed in filename.
     /// Returns false, if the file could not be read.
     bool setExcludeIDs(const char* filename);
-    
+
     /// When reading records, keep only variants with the ids specified
     /// in the passed in filename.
     /// Returns false, if the file could not be read.
     bool setIncludeIDs(const char* filename);
     //@}
 
-    
+
     ///////////////////////
     /// @name  Check if a record should be kept.
     //@{
