From 3dcfd517307248c732d4a7b5214a0c5e9e071f7d Mon Sep 17 00:00:00 2001
From: Daniel Nachun <daniel.nachun@gmail.com>
Date: Thu, 13 Mar 2025 14:33:01 -0700
Subject: [PATCH] make shared library

---
 Makefile | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 4929693..61b4023 100644
--- a/Makefile
+++ b/Makefile
@@ -8,6 +8,11 @@ OBJS=		kthread.o misc.o \
 			unitig.o mag.o bubble.o f_ksw.o
 PROG=		fml-asm
 LIBS=		-lm -lz -lpthread
+ifeq ($(shell uname -s),Linux)
+	LFLAGS = -shared
+else
+	LFLAGS = -dynamiclib -Wl,-install_name,@rpath/libfml.dylib
+endif
 
 .SUFFIXES:.c .o
 
@@ -16,11 +21,11 @@ LIBS=		-lm -lz -lpthread
 
 all:$(PROG)
 
-fml-asm:libfml.a example.o
+fml-asm:libfml$(SHLIB_EXT) example.o
 		$(CC) $(CFLAGS) $^ -o $@ -L. -lfml $(LIBS)
 
-libfml.a:$(OBJS)
-		$(AR) -csru $@ $(OBJS)
+libfml$(SHLIB_EXT):$(OBJS)
+		$(CC) $(CFLAGS) $(LFLAGS) -o $@ -lz $(OBJS)
 
 clean:
 		rm -fr gmon.out *.o ext/*.o a.out $(PROG) *~ *.a *.dSYM session*
-- 
2.47.1

