From 64bbb733ac3459f4b8c2e8e26ff832281e943bc9 Mon Sep 17 00:00:00 2001
From: danielnachun <daniel.nachun@gmail.com>
Date: Wed, 27 Mar 2024 19:00:21 -0700
Subject: [PATCH] use sse2neon

---
 src/abismal/src/abismal.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/abismal/src/abismal.cpp b/src/abismal/src/abismal.cpp
index 59e1479..3cffefc 100644
--- a/src/abismal/src/abismal.cpp
+++ b/src/abismal/src/abismal.cpp
@@ -41,6 +41,9 @@
 #include "sam_record.hpp"
 #include "smithlab_os.hpp"
 #include "smithlab_utils.hpp"
+#ifdef __arm64__
+#include "sse2neon/sse2neon.h"
+#endif
 
 using std::begin;
 using std::cerr;
-- 
2.44.0

