diff --git a/inst/tinytest/test_tiledbarray.R b/inst/tinytest/test_tiledbarray.R
index e50d8720..99875684 100644
--- a/inst/tinytest/test_tiledbarray.R
+++ b/inst/tinytest/test_tiledbarray.R
@@ -1,6 +1,10 @@
 library(tinytest)
 library(tiledb)

+# Skip for conda osx-64 build (controlled by Jinja selectors in recipe) due to
+# spurious test failures
+exit_file("skip for conda osx-64 build")
+
 isMacOS <- (Sys.info()['sysname'] == "Darwin")

 ctx <- tiledb_ctx(limitTileDBCores())
