diff --git a/runtime/dynlink.c b/runtime/dynlink.c
index b92d5e3..9f85d24 100644
--- a/runtime/dynlink.c
+++ b/runtime/dynlink.c
@@ -82,7 +82,6 @@ CAMLexport char_os * caml_get_stdlib_location(void)
   char_os * stdlib;
   stdlib = caml_secure_getenv(T("OCAMLLIB"));
   if (stdlib == NULL) stdlib = caml_secure_getenv(T("CAMLLIB"));
-  if (stdlib == NULL) stdlib = OCAML_STDLIB_DIR;
   return stdlib;
 }
 
diff --git a/stdlib/Makefile b/stdlib/Makefile
index df34bc2..c7936ed 100644
--- a/stdlib/Makefile
+++ b/stdlib/Makefile
@@ -131,17 +131,12 @@ CAMLHEADERS =\
 # patterns and $(subst ...) since `camlheader%:` wouldn't match `camlheader`
 ifeq "$(SHEBANGSCRIPTS)" "true"
 camlhead%: $(ROOTDIR)/Makefile.config Makefile
-ifeq "$(LONG_SHEBANG)" "true"
-	echo '#!/bin/sh' > $@
-	echo 'exec "$(BINDIR)/ocamlrun$(subst er,,$*)" "$$0" "$$@"' >> $@
-else
-	echo '#!$(BINDIR)/ocamlrun$(subst er,,$*)' > $@
-endif
+	echo '#!/usr/bin/env ocamlrun$(subst er,,$*)' > $@
 
 # TODO This does not take long shebangs into account (since TARGET_BINDIR is not
 #      yet processed by configure)
 target_%: $(ROOTDIR)/Makefile.config Makefile
-	echo '#!$(TARGET_BINDIR)/ocamlrun$(subst camlheader,,$*)' > $@
+	echo '#!/usr/bin/env ocamlrun$(subst camlheader,,$*)' > $@
 
 camlheader_ur: Makefile
 	echo '#!' | tr -d '\012' > $@
diff --git a/tools/ocamlmklib.ml b/tools/ocamlmklib.ml
index 0ca7b80..74194e1 100644
--- a/tools/ocamlmklib.ml
+++ b/tools/ocamlmklib.ml
@@ -32,8 +32,7 @@ let mklib out files opts =
 
 (* PR#4783: under Windows, don't use absolute paths because we do
    not know where the binary distribution will be installed. *)
-let compiler_path name =
-  if Sys.os_type = "Win32" then name else Filename.concat Config.bindir name
+let compiler_path name = name
 
 let bytecode_objs = ref []  (* .cmo,.cma,.ml,.mli files to pass to ocamlc *)
 and native_objs = ref []    (* .cmx,.ml,.mli files to pass to ocamlopt *)
diff --git a/utils/config.mlp b/utils/config.mlp
index 44c6ff8..feb25f5 100644
--- a/utils/config.mlp
+++ b/utils/config.mlp
@@ -19,7 +19,7 @@ let version = Sys.ocaml_version
 
 let bindir = "%%BINDIR%%"
 
-let standard_library_default = "%%LIBDIR%%"
+let standard_library_default = (Sys.getenv "OCAML_PREFIX") ^ "/lib/ocaml"
 
 let standard_library =
   try
