From ad601df38cb1f19e020f4d48817bc66f7430248c Mon Sep 17 00:00:00 2001
From: danielnachun <daniel.nachun@gmail.com>
Date: Sun, 4 Feb 2024 11:22:52 -0800
Subject: [PATCH 2/4] handle undefined constants in mmap.c

---
 linux-user/mmap.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/linux-user/mmap.c b/linux-user/mmap.c
index 96c9433e2..01873204f 100644
--- a/linux-user/mmap.c
+++ b/linux-user/mmap.c
@@ -17,6 +17,7 @@
  *  along with this program; if not, see <http://www.gnu.org/licenses/>.
  */
 #include "qemu/osdep.h"
+#include <linux/mman.h>
 #include <sys/shm.h>
 #include "trace.h"
 #include "exec/log.h"
-- 
2.43.0

